jib-cli/src/main/java/com/google/cloud/tools/jib/cli/buildfile/BuildFileSpec.java

```diff
@@ -0,0 +1,197 @@
+/*
+ * Copyright 2020 Google LLC.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package com.google.cloud.tools.jib.cli.buildfile;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.google.cloud.tools.jib.api.Ports;
+import com.google.cloud.tools.jib.api.buildplan.AbsoluteUnixPath;
+import com.google.cloud.tools.jib.api.buildplan.ImageFormat;
+import com.google.cloud.tools.jib.api.buildplan.Port;
+import com.google.common.base.Preconditions;
+import java.time.Instant;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.Set;
+import java.util.stream.Collectors;
+import javax.annotation.Nullable;
+
+/**
+ * A yaml block for specifying a jib cli buildfile.
+ *
+ * <p>Example use of this yaml.
+ *
+ * <pre>{@code
+ * apiVersion: v1alpha1
+ * kind: BuildFile
+ * from: see {@link BaseImageSpec}
+ * creationTime: 100
+ * format: docker
+ * environment:
+ *   env_key: env_value
+ * labels:
+ *   label_key: label_value
+ * volumes:
+ *   - /my/volume
+ * exposedPorts:
+ *   - 8080
+ * user: username
+ * workingDirectory: /workspace
+ * entrypoint:
+ *   - java
+ *   - -jar
+ * cmd:
+ *   - myjar.jar
+ * layers: see {@link LayersSpec}
+ * }</pre>
+ */
+public class BuildFileSpec {
+  private String apiVersion;
+  private String kind;
+  @Nullable private BaseImageSpec from;
+  @Nullable private Instant creationTime;
+  @Nullable private ImageFormat format;
+  @Nullable private Map<String, String> environment;
+  @Nullable private Map<String, String> labels;
+  @Nullable private Set<AbsoluteUnixPath> volumes;
+  @Nullable private Set<Port> exposedPorts;
+  @Nullable private String user;
+  @Nullable private AbsoluteUnixPath workingDirectory;
+  @Nullable private List<String> entrypoint;
+  @Nullable private List<String> cmd;
+
+  @Nullable private LayersSpec layers;
+
+  /**
+   * Constructor for use by jackson to populate this object.
+   *
+   * @param apiVersion the api version of this buildfile
+   * @param kind the kind (always BuildFile)
+   * @param from a {@link BaseImageSpec} for base image
```

Comment:
is it possible to get a bit more context into these two variables?

Reply:
kind and from?